src/qt/bitcoinunits.cpp

```diff
@@ -248,3 +248,39 @@ CAmount BitcoinUnits::maxMoney()
 {
     return MAX_MONEY;
 }
+
+namespace {
+qint8 ToQuint8(BitcoinUnit unit)
+{
+    switch (unit) {
+    case BitcoinUnits::BTC: return 0;
+    case BitcoinUnits::mBTC: return 1;
+    case BitcoinUnits::uBTC: return 2;
+    case BitcoinUnits::SAT: return 3;
+    } // no default case, so the compiler can warn about missing cases
+    assert(false);
+}
+
+BitcoinUnit FromQuint8(qint8 num)
+{
+    switch (num) {
+    case 3: return BitcoinUnits::SAT;
+    case 2: return BitcoinUnits::uBTC;
+    case 1: return BitcoinUnits::mBTC;
+    default: return BitcoinUnits::BTC;
```

Comment:
77edf1bd0149bd2625dff563f3bd4032a12a2233

Add case for `0` and an `assert(false)` in default?

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/gui/pull/60#issuecomment-823554773).